### PR TITLE
build: add tt weak dependency

### PR DESCRIPTION
**Pull request by psergee**

### CODE REVIEW

File: changelogs/unreleased/no-gh-add-tt-dependency.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## feature/build
+
+* Added tt weak dependency
```
Comment (Totktonada) [inline changelogs/unreleased/no-gh-add-tt-dependency.md:3]: I would like to see an issue with the motivation behind the change.

File: rpm/tarantool.spec [hunk 2]
```diff
@@ -130,6 +130,7 @@ Recommends: cmake >= 3.3
 Recommends: make
 Recommends: gcc >= 4.5
 Recommends: gcc-c++ >= 4.5
+Recommends: tt >= 0.4.0
```
[the remark this answers is not in the document]
Comment (psergee) [reply rpm/tarantool.spec:133]: That is why tt is a recommended package: a user can install Tarantool rpm/deb without tt.

File: debian/control [hunk 3]
```diff
@@ -63,7 +63,7 @@ Depends: ${misc:Depends}, adduser, lsb-base,
 # Deps for built-in package manager
 # https://github.com/tarantool/tarantool/issues/2612
  openssl
-Recommends: tarantool-dev, git, build-essential, cmake
+Recommends: tarantool-dev, git, build-essential, cmake, tt (>= 0.4.0)
```
[the remark this answers is not in the document]
Comment (psergee) [reply debian/control:66]: Has the decision to remove recommends/suggests been made yet?

[the remark this answers is not in the document]
Comment (Totktonada) [reply debian/control:66]: Nope. But personally I would at least remove build-essential. It is a lot of packages and the recommended packages are installed by default (if the feature is not disabled explicitly by the system administrator).

File: rpm/tarantool.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (Totktonada) [reply rpm/tarantool.spec:133]: But is it okay to have non-existing package in the recommended ones? In fact, two questions: whether it is okay with `APT::Get::Install-Recommends "false";` and whether it is okay with `APT::Get::Install-Recommends "false";`.

[the remark this answers is not in the document]
Comment (psergee) [reply rpm/tarantool.spec:133]: I think it's worse to fail Tarantool package installation due to strict dependency and missing `tt` package. In my opinion it is okay to have `APT::Install-Recommends` set to `false` by an administrator. Or do you mean that most of server "minimal" distributions have this option set to `false` by default?
@LeonidVas what do you think?

File: changelogs/unreleased/no-gh-add-tt-dependency.md (hunk 1 above)
Comment (psergee) [reply changelogs/unreleased/no-gh-add-tt-dependency.md]: Issue reference is added.

File: rpm/tarantool.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (Totktonada) [reply rpm/tarantool.spec:133]: In this thread my questions are pure technical. Is it okay to have `Recommends: non_existing_package` on Debian based distro with and without `APT::Get::Install-Recommends` enabled. And the same question about RPM based distros.

File: debian/control (hunk 3 above)
[the remark this answers is not in the document]
Comment (Totktonada) [reply debian/control:66]: Well, ok, let's consider it as a separate task.

**Review by LeonidVas — COMMENTED** [review]
Hi! Thank you for the patch.

File: rpm/tarantool.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (psergee) [reply rpm/tarantool.spec:133]: `dnf` installs weak dependencies by default and skips recommendations which are not available.
`apt` behaves in a similar way: installs all available weak deps. Missing recommendations do not fail package installation.
If `APT::Install-Recommends` or `install_weak_deps` are set to `false`, weak dependencies are not installed.

[the remark this answers is not in the document]
Comment (psergee) [reply rpm/tarantool.spec]: Done.

**Review by LeonidVas — APPROVED** [review]
LGTM

File: changelogs/unreleased/no-gh-add-tt-dependency.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+## feature/build
+
+* Added `tt` weak dependency. `tt` command line utility combines
+`tarantoolctl` and `cartridge-cli` functionality into one tool.
+This changes are part of pushing `tt` to public ( tarantool/tt#286 ).
```
[the remark this answers is not in the document]
Comment (Totktonada) [reply changelogs/unreleased/no-gh-add-tt-dependency.md:5]: Nit: This changes -> those changes or this change.

**Review by Totktonada — APPROVED** [review]

**Review by ylobankov — APPROVED** [review]
LGTM

**Comment by Totktonada** [comment]
Pushed to `master` as `3.0.0-entrypoint-106-g10f7109c7`, cherry-picked to `2.11` as `2.11.0-rc1-83-g224b3032b` (will appear in 2.11.1, not 2.11.0, which lives in the `release/2.11.0` branch).

**Pull request merged**